internal/testutil/asserter/assert.go

```diff
@@ -140,14 +154,25 @@ func (a *Asserter) shouldExcludeGroup(group string) bool {
 func entitiesFor(entities []*integration.Entity, pseudotype string) []*integration.Entity {
 	var appropriateEntities []*integration.Entity
 	for _, e := range entities {
-		if strings.Contains(strings.ToLower(e.Metadata.Namespace), strings.ToLower(pseudotype)) {
+		if specGroupNameMatch(e, pseudotype) {
 			appropriateEntities = append(appropriateEntities, e)
 		}
 	}
 
 	return appropriateEntities
 }
 
+// pseudotypeMatch returns true if the specGroupName match with the provided entity.
+func specGroupNameMatch(entity *integration.Entity, specGroupName string) bool {
```

Comment:
I wanted to move the "heuristic" logic to that function, so `entitiesFor` just keeps the entities when `specGroupNameMatch`. So, If we changed the criteria again we would just need to change this new function. That is the reason why it receives the whole entity instead of directly the namespace.

WDYT?

Reply:
I see you point, seems good to me 👍